packages/blockchain/src/consensus/clique.ts

```diff
@@ -0,0 +1,557 @@
+            return voteCMP.equals(vote[1][0])
+          }).length
+          if (num === 0) {
+            beneficiaryVotesDROP.push(vote[1][0])
+          }
+        }
+        let numBeneficiaryVotesDROP = beneficiaryVotesDROP.length
+
+        if (round === 2 && nonce.equals(CLIQUE_NONCE_DROP)) {
+          numBeneficiaryVotesDROP += 1
+        }
+        // Majority consensus
+        if (numBeneficiaryVotesDROP >= limit) {
+          consensus = true
+          // Drop signer
+          activeSigners = activeSigners.filter((signer) => !signer.equals(beneficiary))
+          this._cliqueLatestVotes = this._cliqueLatestVotes.filter(
+            // Discard votes from removed signer and for removed signer
+            (vote) => !vote[1][0].equals(beneficiary) && !vote[1][1].equals(beneficiary)
+          )
+          debug(`[Block ${header.number}] Clique majority consensus (DROP ${beneficiary})`)
+        }
+        if (round === 1) {
+          // Always add the latest vote to the history no matter if already voted
+          // the same vote or not
+          this._cliqueLatestVotes.push(latestVote)
+          debug(
+            `[Block ${header.number}] New clique vote: ${signer} -> ${beneficiary} ${
+              nonce.equals(CLIQUE_NONCE_AUTH) ? 'AUTH' : 'DROP'
+            }`
+          )
+        }
+        if (consensus) {
+          if (round === 1) {
+            debug(
+              `[Block ${header.number}] Clique majority consensus on existing votes -> update signer states`
+            )
+          } else {
+            debug(
+              `[Block ${header.number}] Clique majority consensus on new vote -> update signer states`
+            )
+          }
+          const newSignerState: CliqueSignerState = [header.number, activeSigners]
+          await this.cliqueUpdateSignerStates(newSignerState)
+          return
+        }
+      }
+    }
+
+    // trim to lastEpochBlockNumber - CLIQUE_SIGNER_HISTORY_BLOCK_LIMIT
+    const limit = this.CLIQUE_SIGNER_HISTORY_BLOCK_LIMIT
+    const blockSigners = this._cliqueLatestBlockSigners
+    const lastBlockNumber = blockSigners[blockSigners.length - 1]?.[0]
+    if (lastBlockNumber) {
+      const lastEpochBlockNumber =
+        lastBlockNumber -
+        (lastBlockNumber % BigInt(this.blockchain._common.consensusConfig().epoch))
+      const blockLimit = lastEpochBlockNumber - BigInt(limit)
+      this._cliqueLatestVotes = this._cliqueLatestVotes.filter((state) => state[0] >= blockLimit)
+    }
+
+    // save votes to db
+    const formatted = this._cliqueLatestVotes.map((v) => [
+      bigIntToBuffer(v[0]),
+      [v[1][0].toBuffer(), v[1][1].toBuffer(), v[1][2]],
+    ])
+    await this.blockchain.db.put('CliqueVotes', rlp.encode(formatted), DB_OPTS)
+  }
+
+  /**
+   * Returns a list with the current block signers
+   * (only clique PoA, throws otherwise)
+   */
+  cliqueActiveSigners(): Address[] {
+    this._requireClique()
```

Comment:
I'm not sure if we need this `_requireClique` check anymore since this class can only be initialized when consensus is clique

Reply:
Yeah good call - it's only used in this one spot. Removed